content-moderation/02-video-moderation/03-content-moderation-video-with-image-api.ipynb

```diff
@@ -0,0 +1,897 @@
+    "    result = {\"Timestamp\": float(ts), \"ModerationLabel\": []}\n",
+    "    for l in detectModerationLabelsResponse[\"ModerationLabels\"]:\n",
+    "        result[\"ModerationLabel\"].append(\n",
+    "            {\n",
+    "                \"Confidence\": l[\"Confidence\"],\n",
+    "                \"Name\": l[\"Name\"],\n",
+    "                \"ParentName\": l[\"ParentName\"]\n",
+    "            }\n",
+    "        )\n",
+    "    return result\n",
+    "\n",
+    "# Upload images to s3 and perform moderation, and cleanup temp files on local disk\n",
+    "labels = []\n",
+    "for file in os.listdir(LOCAL_DIR):\n",
+    "    if file.endswith(IMAGE_NAME_EXTENSION):\n",
+    "        # convert file name from sequence to time position\n",
+    "        seq = float(file.replace(IMAGE_NAME_EXTENSION,''))\n",
+    "        ms_pos = 1/SAMPLE_FREQUENCY * (seq-1) * 1000\n",
+    "        s3.upload_file(f'{LOCAL_DIR}/{file}', bucket_name, f'{s3_target_folder}/{ms_pos}.png')\n",
+    "            \n",
+    "        # moderate image\n",
+    "        mr = moderate_image(bucket_name, f'{s3_target_folder}/{ms_pos}.png')\n",
+    "        if mr is not None and len(mr[\"ModerationLabel\"]) > 0:\n",
+    "            labels.append(mr)\n",
+    "            \n",
+    "    # Delete local file: image or video\n",
+    "    os.remove(f'{LOCAL_DIR}/{file}')\n",
+    "\n",
+    "list_temp_s3_prefix.append(s3_target_folder)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "90614d30-c343-4339-b18c-32fdfaf24300",
+   "metadata": {},
+   "source": [
+    "Display the moderation results"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5c428baf-6c71-4eb6-8533-d95bf8a19009",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# sort labels\n",
+    "labels.sort(key=lambda x: x[\"Timestamp\"], reverse=False)\n",
+    "    \n",
+    "result = {\n",
+    "        \"API\": API_NAME,\n",
+    "        \"Video\": {\n",
+    "            \"S3Bucket\": bucket_name,\n",
+    "            \"S3ObjectName\": file_name\n",
+    "        },\n",
+    "        \"ModerationLabels\": labels\n",
+    "    }\n",
+    "\n",
+    "# Display results\n",
+    "print(result)\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ae7798f5-d372-4f6c-925a-b905e18bb443",
+   "metadata": {},
+   "source": [
+    "# Step 4: Clean up <a id=\"step4\"></a>\n",
+    "Clean up sampled images in S3 bucket"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "81c278bd-e099-4fd5-9ddc-4ac5d9b5ab70",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "prefix_to_be_deleted = list(set(list_temp_s3_prefix))\n",
+    "for pf in prefix_to_be_deleted:\n",
+    "    for obj in data_bucket.objects.filter(Prefix=pf):\n",
+    "        s3.delete_object(Bucket=bucket_name, Key=obj.key)"
+   ]
+  }
+ ],
+ "metadata": {
+  "availableInstances": [
```

Comment:
Do we need all this metadata for specifying all the EC2 instances? My preference would be to have the same metadata we have in the other files on the same folder to keep things consistent, unless these are actually mandatory for the lab to work.

Reply:
I've removed EC2 instances in metadata section